lib/LaTeXML/Package/graphics.sty.ltxml

```diff
@@ -244,17 +270,23 @@ sub graphicS_options {
   my $options = ($starred ? ($bb ? "viewport=$bb, clip" : "clip") : '');
   return $options; }
 
+# Combine keyval options into a single options attribute.
 sub graphicX_options {
   my ($starred, $kv) = @_;
-  my $options = ToString($kv);
-  $options .= ',clip=true' if $starred;
-  my $keepaspect;
-  $options .= ',';
-  $keepaspect = 1 if $options =~ s/width=,//;
-  $keepaspect = 1 if $options =~ s/height=,//;
-  if ($keepaspect) { $options .= 'keepaspectratio=true'; }
-  else             { $options =~ s/,$//; }
-  return $options; }
+  my @options = ();
+  my @kv      = $kv->getPairs;
+  push(@options, 'clip=true') if $starred;
+  #  my $keepaspect;
+  my ($saw_w, $saw_h);
+  while (@kv) {
+    my ($key, $value) = (shift(@kv), shift(@kv));
+    $saw_w = 1 if $key =~ /width$/;
+    $saw_h = 1 if $key =~ /height$/;
+    $value = ToString($value);
+    $value =~ s/,/\\,/g;    # Slashify any "," within the value
+    push(@options, $key . '=' . $value); }
+  push(@options, 'keepaspectratio=true') if ($saw_w xor $saw_h) && !$kv->hasKey('keepaspectratio');
```

Comment:
Minor: this `$saw_w xor $saw_h` startled me, so I reread and noticed you could switch to say:
```perl
my $saw_w_or_h = ($key =~ /(?:width|height)$/);
```
and then use that directly (as the individual cases are never used on their own).

Reply:
but that's `or`, not `xor`... or did some bug creep in?

Reply:
It's not a bug, it's a feature :> since both are matching for an exact substring at the end of the same string, it is not theoretically possible for both conditions to be true at the same time.